src/main/kotlin/no/nav/medlemskap/common/Metrics.kt

```diff
@@ -155,4 +153,8 @@ fun totalGauge(): AtomicInteger =
             Gauge.builder("health_check_status") { this }
                     .description("Indikerer applikasjonens helsestatus. 0 er OK, 1 indikerer feil.")
                     .register(Metrics.globalRegistry)
-        }
\ No newline at end of file
+        }
+
+object YtelseSingleton {
```

Comment:
Dette så umiddelbart ikke så veldig trådsikkert ut? Er tanken at du skriver over `var ytelse` for hvert kall? Hva da hvis sykepenger og dagpenger kaller oss ~samtidig?

Reply:
Jeg antar at du prøver å løse dette problemet: https://stackoverflow.com/questions/55373266/how-to-make-request-bound-data-globally-available-in-ktor

Reply:
Jeg tror det er mulig å få til dette med Ktor, men det er nok litt krøkkete: https://blog.tpersson.io/2018/04/22/emulating-request-scoped-objects-with-kotlin-coroutines/
Vi stod overfor dette problemet da vi skulle videresende callId som vi potensielt har mottatt fra sykepenger, og gjenbruke den når vi kaller baksystemene våre. Det løste Richard med å sende callId eksplisitt rundt, som du kan se i koden. Det er nok løsningen som er mest "rett-etter-nesa". Ytelse er vel lagret i Datagrunnlag, som sendes rundt omkring, kan du hente det ut derfra og eksplisitt bruke det inn i metrikkene?

Reply:
Det har du rett i. Ok, ja, virker som det beste er å sende med ytelse eksplisitt

Reply:
Rettet. Ser ikke like elegant ut, men det burde hvertfall være trådsikkert.

Reply:
Enig.. :)